CRM/PivotData/AbstractData.php

```diff
@@ -104,13 +111,72 @@ public function __construct($name, $apiEntityName = NULL) {
     $this->name = $name;
     $this->apiEntityName = $apiEntityName ? $apiEntityName : $name;
 
+    $this->loadLabelCustomizations();
     $dateFields = $this->getDateFields();
 
     foreach ($dateFields as $field) {
       $this->additionalHeaderFields[$field . ' (' . ts('per month') . ')'] = '';
     }
   }
 
+  /**
+   * Loads customized labels from json configuration files.
+   */
+  private function loadLabelCustomizations() {
+    $className = strtr(get_class($this), array('CRM_PivotData_' => ''));
+    $customLabelsFileName = $this->getDefaultResourcesPath() . 'custom_labels/' . $className . '.json';
+
+    if(file_exists($customLabelsFileName)) {
+      $jsonData = file_get_contents($customLabelsFileName);
+      $this->customLabels = json_decode($jsonData, true);
+    }
+  }
+
+  /**
+   * Returns default resources path of the extension.
+   *
+   * @return string
+   */
+  private function getDefaultResourcesPath() {
+    return realpath(__DIR__ . '/../../') . '/resources/';
+  }
+
+  /**
+   * Replaces fields' titles with customized labels.
+   *
+   * @param array $fields
+   *
+   * @return array
+   */
+  protected function replaceCustomLabelsForFieldTitles(&$fields) {
+    foreach ($fields as $key => $field) {
+      if (is_array($field)) {
+        $title = CRM_Utils_Array::value('title', $field, '');
+        $fields[$key]['title'] = $this->replaceCustomLabels($title);
+      } else {
+        $fields[$key] = $this->replaceCustomLabels($field);
+      }
+    }
+  }
+
+  /**
+   * Searches label in customized labels array and returns mapped string value
+   * if it is found. Otherwise returns same label.
+   *
+   * @param string $label
+   *
+   * @return string
+   */
+  protected function replaceCustomLabels($label) {
```

Comment:
the name should be **replaceCustomLabel** instead of **replaceCustomLabels** since this only replace a single label.